File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
       int[] arr=new int[2];
        for(int i=0;i<n;i++){
            for(int j=0;j<n;j++){
                if(i!=j){
                if(nums[i]+nums[j]==target){
                    arr[0]=i;
                    arr[1]=j;         
                break;
                 }
            }
            }
        }
        return arr;
    }
}

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int>vec;
        // for(int i=0;i<nums.size();i++){
        //     for(int j=i+1;j<nums.size();j++){
        //         if(nums[i]+nums[j]==target) {
        //             vec.push_back(i);vec.push_back(j);
        //             break;
        //         }
        //     }
        // }
        // return vec;
        
        vector<int> v;
        unordered_map<int,int> mp;
        for(int i=0;i<nums.size();i++){
            if(mp.find(target-nums[i])!=mp.end()){
                v.push_back(mp[target-nums[i]]);
                v.push_back(i);
                return v;
            }
            mp[nums[i]]=i;
        }
        return v;
    }
};